browser/brave_shields/brave_shields_util_profiles_unittest.cc

```diff
@@ -0,0 +1,212 @@
+               << " Expected " << expect << "\n";
+      }
+
+      auto incognito = incognito_profile();
+      // Now change value for incognito and expect that values are not depended
+      // on the regular profile.
+      for (const auto& [ivalue, iexpect] : cases) {
+        setter(hcsm(incognito), ivalue);
+        const auto iget = getter(hcsm(incognito));
+        if (iexpect != iget) {
+          issues << "Incognito profile: Set " << ivalue << " Get " << iget
+                 << " Expected " << iexpect << " Regular profile value "
+                 << value << "\n";
+        }
+      }
+    }
+    const auto message = issues.GetString();
+    EXPECT_TRUE(message.empty()) << message;
+  }
+
+ private:
+  content::BrowserTaskEnvironment task_environment_;
+  TestingProfile profile_;
+  ScopedTestingLocalState local_state_;
+};
+
+TEST_F(BraveShieldsUtilProfilesTest, SetBraveShieldsEnabled) {
+  constexpr std::pair<bool, bool> kExpects[] = {{true, true}, {false, false}};
+
+  auto set = [this](HostContentSettingsMap* map, bool value) {
+    SetBraveShieldsEnabled(map, value, test_url);
+  };
+  auto get = [this](HostContentSettingsMap* map) {
+    return GetBraveShieldsEnabled(map, test_url);
+  };
+
+  RunTest<bool>(kExpects, std::move(set), std::move(get));
+}
+
+TEST_F(BraveShieldsUtilProfilesTest, SetAdControlType) {
+  constexpr std::pair<ControlType, ControlType> kExpects[] = {{ALLOW, ALLOW},
+                                                              {BLOCK, BLOCK}};
+
+  auto set = [this](HostContentSettingsMap* map, ControlType value) {
+    SetAdControlType(map, value, test_url);
+  };
+  auto get = [this](HostContentSettingsMap* map) {
+    return GetAdControlType(map, test_url);
+  };
+
+  RunTest<ControlType>(kExpects, std::move(set), std::move(get));
+}
+
+TEST_F(BraveShieldsUtilProfilesTest, SetCosmeticFilteringControlType) {
+  constexpr std::pair<ControlType, ControlType> kExpects[] = {
+      {ALLOW, ALLOW}, {BLOCK, BLOCK}, {BLOCK_THIRD_PARTY, BLOCK_THIRD_PARTY}};
+
+  auto set = [this](HostContentSettingsMap* map, ControlType value) {
+    SetCosmeticFilteringControlType(map, value, test_url);
+  };
+  auto get = [this](HostContentSettingsMap* map) {
+    return GetCosmeticFilteringControlType(map, test_url);
+  };
+
+  RunTest<ControlType>(kExpects, std::move(set), std::move(get));
+}
+
+TEST_F(BraveShieldsUtilProfilesTest, SetCookieControlType) {
+  constexpr std::pair<ControlType, ControlType> kExpects[] = {
+      {ALLOW, ALLOW}, {BLOCK, BLOCK}, {BLOCK_THIRD_PARTY, BLOCK_THIRD_PARTY}};
+
+  auto set = [this](HostContentSettingsMap* map, ControlType value) {
+    SetCookieControlType(map, regular_profile()->GetPrefs(), value, test_url);
+  };
+
+  auto cookie_settings =
+      CookieSettingsFactory::GetForProfile(regular_profile());
+
+  auto get = [this, cookie_settings](HostContentSettingsMap* map) {
+    return GetCookieControlType(map, cookie_settings.get(), test_url);
+  };
```

Comment:
Do you think it would be better if we avoid the capture of `cookie_settings` and do just:

```cxx
  auto get = [this](HostContentSettingsMap* map) {
    return GetCookieControlType(map, CookieSettingsFactory::GetForProfile(regular_profile()).get(), test_url);
  };
```